controllers/provider-packet/pkg/packet/client/client.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"strings"
+
+	"github.com/packethost/packngo"
+)
+
+type packetClient struct {
+	packet *packngo.Client
+}
+
+// NewClient creates a new Client for the given Packet credentials
+func NewClient(packetAPIKey string) ClientInterface {
```

Comment:
Is this still WIP?

Reply:
No, see gardener/gardener#916, especially https://github.com/gardener/gardener/pull/916#discussion_r281515425. According to this, @deitch wants to keep the client in case he wants to use it in the future.

As he will take over ownership of the packet provider I just added it here as well.